DMX.Core.Api.Tests.Unit/Services/Foundations/LabCommands/LabCommandServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,202 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using DMX.Core.Api.Models.Foundations.LabCommands;
+using DMX.Core.Api.Models.Foundations.LabCommands.Exceptions;
+using FluentAssertions;
+using Force.DeepCloner;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Foundations.LabCommands
+{
+    public partial class LabCommandServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnModifyIfSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+            LabCommand someLabCommand = CreateRandomLabCommand(randomDateTimeOffset);
+            int randomNumber = GetRandomNumber();
+            someLabCommand.UpdatedDate = someLabCommand.CreatedDate.AddSeconds(randomNumber);
+            SqlException sqlException = GetSqlException();
+
+            var failedLabCommandStorageException =
+                new FailedLabCommandStorageException(sqlException);
+
+            var expectedLabCommandDependencyException =
+                new LabCommandDependencyException(failedLabCommandStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>

```

Comment:
If you hook up the `SqlException` to the `dateTimeBroker` you wouldn't need to add all that stuff at the top to bypass the validation rules. do the same for the rest.

Reply:
Ahh I guess this comes back to a discussion we had a while ago that the exception tests' purpose is solely to test _how_ we handle specific exception types, rather than _where_ those exception types would logically be occurring. Updating!